nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/FetchSlack.java

```diff
@@ -0,0 +1,295 @@
+    .description("Token for slack to use when downloading attachments.")
+    .required(true)
+    .sensitive(true)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
+
+  static final Relationship REL_SUCCESS_REQ = new Relationship.Builder()
+    .name("Original")
+    .description("The original FlowFile will be routed upon success (2xx status codes)")
+    .build();
+
+  static final Relationship REL_RESPONSE = new Relationship.Builder()
+    .name("Response")
+    .description("A Response FlowFile will be routed upon success (2xx status codes)")
+    .build();
+
+  static final Relationship REL_RETRY = new Relationship.Builder()
+    .name("Retry")
+    .description("The original FlowFile will be routed on any status code that can be retried (5xx status codes)")
+    .build();
+
+  static final Relationship REL_NO_RETRY = new Relationship.Builder()
+    .name("No Retry")
+    .description("The original FlowFile will be routed on any status code that should NOT be retried (1xx, 3xx, 4xx status codes)")
+    .build();
+
+  static final Relationship REL_FAILURE = new Relationship.Builder()
+    .name("Failure")
+    .description("The original FlowFile will be routed on any type of connection failure, timeout or general exception.")
+    .build();
+
+  private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+    REL_SUCCESS_REQ, REL_RESPONSE, REL_RETRY, REL_NO_RETRY, REL_FAILURE)));
+
+  private static final List<PropertyDescriptor> DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+    API_TOKEN));
+
+  @Override
+  public Set<Relationship> getRelationships() {
+    return RELATIONSHIPS;
+  }
+
+  @Override
+  public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+    return DESCRIPTORS;
+  }
+
+  @OnScheduled
+  public void createClient(final ProcessContext context) {
+    Header header = new BasicHeader(
+      HttpHeaders.AUTHORIZATION, "Bearer " + context.getProperty(API_TOKEN).getValue());
+    List<Header> headers = Collections.singletonList(header);
+    client = HttpClients.custom().setDefaultHeaders(headers).build();
+  }
+
+  @OnUnscheduled
+  @OnShutdown
+  public void releaseClient() {
+    try {
+      client.close();
+    } catch (IOException e) {
+      getLogger().warn("Error while closing HttpClient", e);
+    }
+    client = null;
+  }
+
+  @Override
+  public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+
+    FlowFile requestFlowFile = session.get();
+    if (requestFlowFile == null) {
+      context.yield();
+
+      return;
+    }
+
+    try {
+      List<JsonObject> filesToDownload;
+      try (InputStream requestInputStream = session.read(requestFlowFile)) {
+        filesToDownload = collectFilesToDownload(requestInputStream);
+      }
+      if (filesToDownload == null || filesToDownload.isEmpty()) {
+        session.transfer(requestFlowFile, REL_FAILURE);
```

Comment:
I think we shouldn't route incoming FlowFiles to failure in this case. How about route it to 'original'?